frontend/src/app/service/transformation/transformation-action.service.spec.ts

```diff
@@ -1,36 +1,174 @@
+        },
+        {
+          provide: NotificationService,
+          useValue: notificationService
         }
       ]
     });
-    transformationActionService = TestBed.inject(TransformationActionService);
   });
 
-  it('should be created', () => {
+  beforeEach(() => {
+    const matDialog = TestBed.inject(MatDialog);
+    const mockStore = TestBed.inject(MockStore);
+    transformationHttpService = TestBed.inject(TransformationHttpService);
+    transformationService = TestBed.inject(TransformationService);
+    tabItemService = TestBed.inject(TabItemService);
+    notificationService = TestBed.inject(NotificationService);
+    transformationActionService = new TransformationActionServiceExtended(
+      matDialog,
+      mockStore,
+      transformationHttpService,
+      transformationService,
+      tabItemService,
+      notificationService
+    );
+  });
+
+  it('TransformationActionService should be created', () => {
+    // Assert
     expect(transformationActionService).toBeTruthy();
   });
+
+  it('Copy transformation, type component', () => {
+    // Arrange
+    const mockNewId = 'mockId1';
+    const mockGroupId = 'mockGroupId1';
+    const mockSuffix = 'Copy';
+    // Act
+    const copyTransformation = transformationActionService.copyTransformation(
+      mockNewId,
+      mockGroupId,
+      mockSuffix,
+      mockTransformation
+    );
+    // Assert
+    expect(copyTransformation.id).toBe('mockId1');
+    expect(copyTransformation.revision_group_id).toBe('mockGroupId1');
+    expect(copyTransformation.version_tag).toBe('0.0.1 Copy');
+    expect(copyTransformation.state).toBe(RevisionState.DRAFT);
+
+    expect(copyTransformation.io_interface.inputs[0].id).not.toBe(
+      'mockInputId0'
+    );
+    expect(copyTransformation.io_interface.inputs[0].name).toBe('mockInput');
+    expect(copyTransformation.io_interface.inputs[0].data_type).toBe(
+      IOType.ANY
+    );
+
+    expect(copyTransformation.io_interface.outputs[0].id).not.toBe(
+      'mockOutputId0'
+    );
+    expect(copyTransformation.io_interface.outputs[0].name).toBe('mockOutput');
+    expect(copyTransformation.io_interface.outputs[0].data_type).toBe(
+      IOType.ANY
+    );
+  });
+
+  it('Copy transformation, type workflow', () => {
+    // Arrange
+    const mockNewId = 'mockId1';
+    const mockGroupId = 'mockGroupId1';
+    const mockSuffix = 'Copy';
+    mockTransformation.type = TransformationType.WORKFLOW;
+    // Act
+    const copyTransformation = transformationActionService.copyTransformation(
+      mockNewId,
+      mockGroupId,
+      mockSuffix,
+      mockTransformation
+    );
+    // Assert
+    expect(copyTransformation.id).toBe('mockId1');
+    expect(copyTransformation.revision_group_id).toBe('mockGroupId1');
+    expect(copyTransformation.version_tag).toBe('0.0.1 Copy');
+    expect(copyTransformation.state).toBe(RevisionState.DRAFT);
+
+    expect(copyTransformation.io_interface.inputs).toHaveSize(0);
```

Comment:
expect().withContext('inputs in io interface should be empty because they are regenerated in the backend')